### PR TITLE
Sentiment analysis 4

### PR DESCRIPTION
**Pull request by anooj-shah**
Add sentiment score to page. Display n/a if null.
Change content-display method from collaborating with Kiba. This was done to remove the use of innerHTML. However, I still use it to initialize the messagesElement. Is this fine?
The logic was that I could make the sentiment score -2 if it is null. Then in script.js, I could print 'n/a' on the page to indicate that it is not available.
Screen Shot 2020-03-06 at 2 35 23 PM

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -1,24 +1,21 @@
...
     messagesElement.innerHTML = "";
     console.log(messages);
     for (i of messages) {
-      messagesElement.innerHTML += "<p>"+i.message+"</p><br>";
+      if (i.score == -2) {
+        messagesElement.appendChild(createListElement(i.message + " Score: n/a"));
+      }
+      else {
+        messagesElement.appendChild(createListElement(i.message + " Score: " + i.score));
+      }
```
Comment (classm-sps) [inline portfolio/src/main/webapp/script.js]: One principle to try and follow is to reduce (as much as possible) the amount of duplicated code. Note how lines 8 and 11 are almost identical: `messagesElement.appendChild(createListElement(i.message + <something>))`. So one thing to consider is could we precompute the `<something>` and then only do the `appendChild` call once? I'd do something like this:

```
const printableScore = (i.score == -2 ? "n/a" : i.score);
messages.appendChild(createListElement(i.message + " Score: " + printableScore));
```

(Note: Javascript is not my primary language, but I'm certain that something like this is possible).

Why is this principle a good idea? It generally leads to code that is more easily maintained. Imagine if you wanted to change the text to be "The score for this message is: "...then this change must be made accurately in two places, which is error prone (could be forgotten, could introduce typos, etc.). Another way of looking at it is "a single thing is always consistent with itself)".

**Review by classm-sps — CHANGES_REQUESTED** [review]
I'm really glad you were able to learn from Kiba! This looks great, just one comment.

**Comment by classm-sps** [comment]
Oh and re: your question about using `.innerHTML` to clear out the field, that's totally fine. The issue is if you're using `.innerHTML` to render content that you don't control (e.g., text entered by potentially malicious site visitors).

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (anooj-shah) [reply portfolio/src/main/webapp/script.js]: Very good point!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -36,7 +36,8 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  // Using '2' to indicate an invalid sentiment score 
+  private static final int INVALID_SENTIMENT_SCORE = 2;
```
Comment (classm-sps) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:41]: please add a blank line to separate the constant declaration from the method declaration.

**Review by classm-sps — APPROVED** [review]

**Pull request merged**